diesel_cli/src/infer_schema_internals/sqlite.rs

```diff
@@ -141,7 +142,10 @@ pub fn get_table_data(
     } else {
         format!("PRAGMA TABLE_INFO('{}')", &table.sql_name)
     };
-    let mut result = sql::<pragma_table_info::SqlType>(&query).load(conn)?;
+
+    // See: https://github.com/diesel-rs/diesel/issues/3579 as to why we use a direct
+    // `sql_query` with `QueryableByName` instead of using `sql::<pragma_table_info::SqlType>`.
+    let mut result = sql_query(query).load::<ColumnInformation>(conn)?;
```

Comment:
I think the `table!` definition of `pragma_table_info` can now be removed. It's not used anymore, beside in the implementation of `QueryableByName` and there we can just use the sql types from the table definition.